kube.libsonnet

```diff
@@ -706,4 +706,34 @@
       podSelector: {},
     },
   },
+
+  VerticalPodAutoscaler(name):: $._Object("autoscaling.k8s.io/v1beta2", "VerticalPodAutoscaler", name) {
+    spec: {
+      local spec = self,
+      targetRef_:: null,
+      targetRef: if spec.targetRef_ != null then {
+        apiVersion: spec.targetRef_.apiVersion,
+        kind: spec.targetRef_.kind,
+        name: spec.targetRef_.metadata.name,
+      } else {},
+      updatePolicy: {
+        updateMode: "Auto",
+      },
+      assert self.targetRef != {} : "VerticalPodAutoscaler: must set a targetRef",
+    },
+  },
+  // Helper function to ease VPA creation as e.g.:
+  // foo_vpa:: kube.createVPAFor($.foo_deploy)
+  createVPAFor(target, mode="Auto"):: $.VerticalPodAutoscaler(target.metadata.name) {
+    metadata+: {
+      namespace: target.metadata.namespace,
+      labels+: target.metadata.labels,
+    },
+    spec+: {
+      targetRef_:: target,
+      updatePolicy+: {
+        updateMode: mode,
+      },
+    },
+  },
```

Comment:
I think calling this "targetRef" is confusing, as this field will not hold an [autoscaling.CrossVersionObjectReference](https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/autoscaling/types.go#L64) object as the `targetRef` field (i.e. an object that has `apiVersion`, `kind` and `name`) but instead it will hold an actual object for which we'll derive the reference.

We already have a construct for that, see `CrossVersionObjectReference` and the use of `target` in https://github.com/bitnami-labs/kube-libsonnet/blob/master/kube.libsonnet#L439

Suggestion: let's do the same as as we did for the HorizontalPodAutoscaler and end up with smth like:

```
  VerticalPodAutoscaler(name):: $._Object("autoscaling.k8s.io/v1beta2", "VerticalPodAutoscaler", name) {
    local vpa = self,

    target:: error "target required",

    spec: {
      targetRef: $.CrossVersionObjectReference(vpa.target),
      updatePolicy: {
        updateMode: "Auto",
      },
    },
  },
  // Helper function to ease VPA creation as e.g.:
  // foo_vpa:: kube.createVPAFor($.foo_deploy)
  createVPAFor(target, mode="Auto"):: $.VerticalPodAutoscaler(target.metadata.name) {
    target:: target,

    metadata+: {
      namespace: target.metadata.namespace,
      labels+: target.metadata.labels,
    },
    spec+: {
      updatePolicy+: {
        updateMode: mode,
      },
    },
  },
```

Reply:
doh `CrossVersionObjectReference` :man_facepalming: , will fix as per your review, thanks !